paasta_tools/kubernetes_tools.py

```diff
@@ -474,20 +475,26 @@ def get_deployment_strategy_config(self) -> V1DeploymentStrategy:
 
         return V1DeploymentStrategy(type=strategy_type, rolling_update=rolling_update)
 
-    def get_sanitised_volume_name(self, volume_name: str) -> str:
+    def get_sanitised_volume_name(self, volume_name: str, length_limit: int = 0) -> str:
         """I know but we really aren't allowed many characters..."""
         volume_name = volume_name.rstrip("/")
         sanitised = volume_name.replace("/", "slash-").replace(".", "dot-")
-        return sanitise_kubernetes_name(sanitised)
+        sanitised_name = sanitise_kubernetes_name(sanitised)
+        if length_limit and len(sanitised_name) > length_limit:
+            sanitised_name = (
+                sanitised_name[0 : length_limit - 4]
+                + hashlib.md5(sanitised_name.encode("ascii")).hexdigest()[:4]
+            )
+        return sanitised_name
 
     def get_docker_volume_name(self, docker_volume: DockerVolume) -> str:
         return self.get_sanitised_volume_name(
-            "host--{name}".format(name=docker_volume["hostPath"])
+            "host--{name}".format(name=docker_volume["hostPath"], length_limit=63)
```

Comment:
(here and in pv name) seems you're passing `length_limit` to `.format()` instead of `get_sanitised_colume_name`?

Reply:
Nice catch, thanks.